### DISCUSSION TITLE
Mapping of GPX fields from external file that used by gopro-dashboard.py

### DISCUSSION THREAD

**Discussion by mishuha** [time4tea/gopro-dashboard-overlay#118] [Q&A]
I have my own software to process GoPro metadata. There I'm doing filtering and other stuff to get more precise dataset stored in CSV. I convert it to GPX and then feed it to dashboard-overlay. And for now I can't pass 2 items: DOP and speed.

For the moment I'm trying to export DOP and speed to GPX. I have a GPX file consisting full of non-fixed data. 
 According to [www.topografix.com/GPX/1/1/](url) there are options:
```
<xsd:element name="hdop" type="xsd:decimal" minOccurs="0"/>
<xsd:element name="vdop" type="xsd:decimal" minOccurs="0"/>
<xsd:element name="pdop" type="xsd:decimal" minOccurs="0"/>
```
I have tried to rename field using each of them with value 9999 but every time got "fix=3d" with widget `<component type="gps-lock-icon""/>`. Widget "dop" shows value = 10.

1. What DOP fieldname is used in GPX file? And is it used at all? Without this `--gps-dop-max` seems completely useless for external data.

2. As for field "speed" it was supported by GPX1.0, but seems lost in GPX1.1. Is there a way to pass it to script via GPX?
 
3. Which fields are actually supported from GPX schema?

**Comment by time4tea** [comment]
Hi,

What is it about the in-built GoPro parsing that means that you though it was worth writing your own? - I don't really have a plan to support all GPX fields at the moment.

Cheers,

**Comment by mishuha** [comment]
Hi,

Because of this:
![clip1](https://user-images.githubusercontent.com/53226085/224538386-376f55a3-e031-4b05-b102-9de82bbdc54f.jpg)
That's a pure gopro source file, single track. A filtering set `--gps-speed-max 50` was applied to in the Dashboard, but on video when I stopped GoPro still continued "moving" me at low speed. Also it was supposed to be a straight line but according to GoPro metadata I was teleporting around the city. My point moved across the map despite that track was not drawn. That things I want to filter out.

Thanks

**Reply by time4tea** [reply]
Hey, thanks for this info. Usually this is down to poor gps reception. You can tell by using gopro-to-csv.py and looking at the DOP value.

There are various filters that exist in the software that can help with this problem, but not really "fix" it.

You already used --gps-speed-max there is also --gps-dop-max 

Using a custom layout you can plot the DOP on the video which can also help diagnose these issues.

Let me know how you get on!

**Reply by mishuha** [reply]
Yes it filters particular files. 
```
NOTE: 49 GoPro GPS readings were mapped to 'NO_LOCK', for the following reasons:
* Heuristics -> 46
* Speed > 50.0 kph -> 3
```
OK, I'd happy to use it but there is the second reason: what if I need to draw an entire travel track summarized from all the files?

I have tried to concatenate files by gopro-join and got readable video and gps data for particular sets of recording. After some naming manipalations I joined them with subsequent recordings. I got an expected gps time gap between joined recordings. 

Feeding this into the gopro-dashboard, made a mess: the date (and data) were out of sync with the video but surprisingly the whole track was successfully drawn to the map. Almost all points were played at the very beginning of the video and rest of them somewhere in random other parts of the video. So I made a conclusion that internal methods are unavailable for such operations. 
Tons of errors like this:
```
Closest item to wanted time Timeunit ms=1605203.6 is Timeunit ms=26669.769 away
Closest item to wanted time Timeunit ms=1605604.5 is Timeunit ms=27070.669 away
Closest item to wanted time Timeunit ms=1606005.4 is Timeunit ms=27471.569 away

```

Despite this overlay generating using external GPX track is just OK. It fills time gaps by creation a lot of extra video data in between. The only thing I have to do in the videoeditor is to cut gaps to sync first GPSU from beginings  of each video with timer from the overlay. That's a little tedious but it's the best way so far.


So this track (45km) was made from 6 sets of recordings:

![clp10](https://user-images.githubusercontent.com/53226085/224774715-6c11d9b4-15a9-4a81-baa2-2c3eb2869e25.jpg)
And I have no idea how to do this using internal functionality

**Comment by mishuha** [comment]
I've just re-read issues and found similar discussions #85, #75, so closing on that point. Thanks for the answers.
